packages/mdc-textfield/README.md

```diff
@@ -132,9 +132,61 @@ To disable the text field, add the `disabled` attribute to the `<input>` element
 ### Text Field with Helper Text
 
 The helper text provides supplemental information and/or validation messages to users. It appears on input field focus
-and disappears on input field blur by default, or it can be persistent.
+and disappears on input field blur by default, or it can be persistent. Helper text should be rendered inside `.mdc-text-field-helper-line` element
+which is immediate sibling of `.mdc-text-field`.
 See [here](helper-text/) for more information on using helper text.
 
+```html
+<div class="mdc-text-field">
+  <input type="text" id="my-text-field" class="mdc-text-field__input">
+  <label class="mdc-floating-label" for="my-text-field">My Label</label>
+  <div class="mdc-line-ripple"></div>
+</div>
+<div class="mdc-text-field-helper-line">
+  <div class="mdc-text-field-helper-text">helper text</div>
+</div>
+```
+
+### Text Field with Character Counter
+
+Character counter is used if there is a character limit. It displays the ratio of characters used and the total character limit.
+Helper text should be rendered inside `.mdc-text-field-helper-line` element which is immediate sibling of `.mdc-text-field`.
+
+```html
+<div class="mdc-text-field">
+  <input type="text" id="my-text-field" class="mdc-text-field__input" maxlength="10">
+  <label class="mdc-floating-label" for="my-text-field">My Label</label>
+  <div class="mdc-line-ripple"></div>
+</div>
+<div class="mdc-text-field-helper-line">
+  <div class="mdc-text-field-character-counter">0 / 10</div>
+</div>
+```
+
+See [here](character-counter/) for more information on using character counter.
```

Comment:
Should this be above or below the code example? It's above for helper text but below for character counter.